node/lifecycle_test.go

```diff
@@ -247,19 +283,51 @@ func wireUpSystem(ctx context.Context, provider PodLifecycleHandler, f testFunct
 
 	// Create the fake client.
 	client := fake.NewSimpleClientset()
+	tracker := client.Tracker()
```

Comment:
It turns out our reactor update / checking logic was incorrect.

Reply:
Can you explain?

Reply:
1. We never checked if the ResourceVersion was set correctly on update (check before set)
2. Although we incremented the ResourceVersion, we never actually did anything with that, since we returned "false" to handled.